src/main/java/io/mosip/mimoto/service/impl/PresentationServiceImpl.java

```
@@ -0,0 +1,93 @@
+
+    @Autowired
+    RestApiClient restApiClient;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Value("${mosip.inji.verify.redirect.url}")
+    String injiVerifyRedirectUrl;
+
+    private final Logger logger = LoggerFactory.getLogger(PresentationServiceImpl.class);
+
+    @Override
+    public String authorizePresentation(PresentationRequestDTO presentationRequestDTO) throws ApiNotAccessibleException, IOException {
+
+        logger.info("Started the presentation Validation");
+        verifiersService.validateVerifier(presentationRequestDTO);
+
+        logger.info("Started the Credential Download From DataShare");
+        String credentials_uri = presentationRequestDTO.getResource();
+        String  vcCredentialResponseString = restApiClient.getApi(credentials_uri, String.class);
+
+        logger.info("Started the ObjectMapping");
+        VCCredentialResponse vcCredentialResponse = objectMapper.readValue(vcCredentialResponseString, VCCredentialResponse.class);
+        PresentationDefinitionDTO presentationDefinitionDTO = objectMapper.readValue(presentationRequestDTO.getPresentation_definition(), PresentationDefinitionDTO.class);
+
+        if(presentationDefinitionDTO.getInputDescriptors().get(0).getFormat().getLdpVc().getProofTypes().get(0).equals(vcCredentialResponse.getCredential().getProof().getType())){
+            logger.info("Started the Construction of VP token");
+            String vpToken = constructVerifiablePresentationString(vcCredentialResponse.getCredential());
+            String presentationSubmission =constructPresentationSubmission(vpToken);
+            return String.format(injiVerifyRedirectUrl,
+                    presentationRequestDTO.getRedirect_uri(),
+                    Base64.getUrlEncoder().encodeToString(vpToken.getBytes(StandardCharsets.UTF_8)),
+                    Base64.getUrlEncoder().encodeToString(presentationSubmission.getBytes(StandardCharsets.UTF_8)));
+        }
+        return null;
```

null response is not handled as error in controller. Should it throw an exception instead of null?

---

modified to throw exception and more exception handling will be handled on a separate story. this story is for the fruitful case.